### DISCUSSION TITLE
Shallow water equations, 2 dimensions, applying a pressure pulse on any one side.

### DISCUSSION THREAD

**Discussion by RohanRajagopal** [CliMA/Oceananigans.jl#2776] [Computational science!]
Good day, I am from a Neuroscience background, I wanted to simulate a shallow water equations model with a pressure pulse added on one side of a bounded container. And in addition adding a droplet of water, maybe something as described in the following link (https://en.wikipedia.org/wiki/File:Shallow_water_waves.gif) 

Unfortunately I am have little to no experience in this field and very limited hands on experience with this great library. Can someone suggest how I could go about setting up the models mentioned above in this library? 

Any help is deeply appreciated.

**Comment by glwagner** [comment]
I'm not sure what you mean by "pressure pulse", but perhaps you can start by setting up an initial value problem. This example illustrates syntax for `ShallowWaterModel`:

https://github.com/CliMA/Oceananigans.jl/blob/main/examples/shallow_water_Bickley_jet.jl

I suggest using a grid that's something like

```julia
grid = RectilinearGrid(size = (Nx, Ny),
                       x = (-5, 5), y = (-5, 5),
                       topology = (Bounded, Bounded, Flat))
```

which is a square, bounded domain. You'll also want to simplify your model from the example; eg you don't want Coriolis forces. You may get away without viscosity / turbulence closure as well, if you keep that model's WENO advection scheme.

To create a wave like in the animation you posted, you can use a Gaussian initial condition, something like

```julia
h0 = 0.1 # bump amplitude
hi(x, y, z) = 1 + h0 * exp(-x^2 + y^2)
set!(model, h=hi)
```

As an exercise, I suggest calculating the gravity wave speed, `c = sqrt(g * h)` where `g` is gravitational acceleration. Then you should use a time-step on the order

```julia
dt = ep * dx / c
```

where `dx` is grid spacing and `ep` is a small number, perhaps 0.01-0.1.

Good luck and have fun!
